messaging/src/test/java/org/axonframework/eventhandling/GapAwareTrackingTokenTest.java

```diff
@@ -247,6 +278,87 @@ void upperBound() {
         assertEquals(GapAwareTrackingToken.newInstance(15, emptyList()), token5.upperBound(token3));
     }
 
+    @Test
+    void upperBoundCase2() {
+
+//        T1 = 7, 8, 10+
+//        T2 = 1+
+//        upper(T1,T2) = 7, 8, 10+  --> token 1 not changed = replay
```

Comment:
This test class shouldn't refer to replay ever, just as replay tokens shouldn't refer to gap aware tokens. :o